zero/src/main/java/mutiny/zero/ZeroPublisher.java

```diff
@@ -0,0 +1,80 @@
+package mutiny.zero;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.reactivestreams.Publisher;
+
+import mutiny.zero.internal.*;
+
+public interface ZeroPublisher {
+
+    // ---- "Iterate over something" ---- //
+
+    @SafeVarargs
+    static <T> Publisher<T> fromItems(T... items) {
+        requireNonNull(items, "The items array cannot be null");
+        return fromIterable(Arrays.asList(items));
+    }
+
+    static <T> Publisher<T> fromIterable(Iterable<T> iterable) {
+        requireNonNull(iterable, "The iterable cannot be null");
+        return new IterablePublisher<>(iterable);
+    }
+
+    static <T> Publisher<T> fromStream(Supplier<Stream<T>> supplier) {
+        requireNonNull(supplier, "The supplier cannot be null");
+        return new StreamPublisher<>(supplier);
+    }
+
+    static <S, T> Publisher<T> fromGenerator(Supplier<S> stateSupplier, Function<S, Iterator<T>> generator) {
+        requireNonNull(stateSupplier, "The state supplier cannot be null");
+        requireNonNull(generator, "The generator supplier cannot be null");
+        return new GeneratorPublisher<>(stateSupplier, generator);
+    }
```

Comment:
There is also the ability to give a generator Supplier<X> which returns a new X for each _request_. IT's a specialized version of the existing `fromGenerator`. I'm not totally sure it's useful, but it's something we often see.